lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1093,6 +1103,19 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
   return success();
 }
 
+LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
+                                     scf::ReduceOp reduceOp) const {
+  if (!reduceOp.getOperands().empty())
```

Comment:
It might be nice here to emit some error message, since this might be a surprising pre-condition.